1_fetch/src/my_happy_script.R

```diff
@@ -1,96 +1,133 @@
+# log x axis vector
 n_profs <- c(2, 10, 50, 100, 500, 980)
 
+
+# set axis, this cannot be a function becuase 
 axis(1, at = c(-100, n_profs, 1e10), labels = c("", n_profs, ""), tck = -0.01)
 axis(2, at = seq(0,10), las = 1, tck = -0.01)
 
 # slight horizontal offsets so the markers don't overlap:
 offsets <- data.frame(pgdl = c(0.15, 0.5, 3, 7, 20, 30)) %>%
   mutate(dl = -pgdl, pb = 0, n_prof = n_profs)
 
+# loop: do this for each type of model in the analysis
 for (mod in c('pb','dl','pgdl')){
-  mod_data <- filter(eval_data, model_type == mod)
-  mod_profiles <- unique(mod_data$n_prof)
-  for (mod_profile in mod_profiles){
-    d <- filter(mod_data, n_prof == mod_profile) %>% summarize(y0 = min(rmse), y1 = max(rmse), col = unique(col))
+  
+  #outer
+  mod_data <- filter(eval_data(getCSV()), model_type == mod) # mod data is set to a smaller dataframe with only 1 type of model
+  mod_profiles <- unique(mod_data$n_prof) # simply a cleaned version, no copies of data points 
+  
+  #inner
+  for (mod_profile in mod_profiles) # what I think this means is for each set of data in the x array of log values
+  {
+    d <- filter(mod_data, n_prof == mod_profile) %>% 
+      summarize(y0 = min(rmse), y1 = max(rmse), col = unique(col))
     x_pos <- offsets %>% filter(n_prof == mod_profile) %>% pull(!!mod) + mod_profile
+    # this is its own thing this line adds the values to a line for the plot
     lines(c(x_pos, x_pos), c(d$y0, d$y1), col = d$col, lwd = 2.5)
   }
-  d <- group_by(mod_data, n_prof) %>% summarize(y = mean(rmse), col = unique(col), pch = unique(pch)) %>%
-    rename(x = n_prof) %>% arrange(x)
   
+  #outer
+  d <- group_by(mod_data, n_prof) %>% 
+    summarize(y = mean(rmse), col = unique(col), pch = unique(pch)) %>%
+    rename(x = n_prof) %>% arrange(x)
+  # adds lines
   lines(d$x + tail(offsets[[mod]], nrow(d)), d$y, col = d$col[1], lty = 'dashed')
+  # adds points
   points(d$x + tail(offsets[[mod]], nrow(d)), d$y, pch = d$pch[1], col = d$col[1], bg = 'white', lwd = 2.5, cex = 1.5)
-  
 }
 
-points(2.2, 0.79, col = '#7570b3', pch = 23, bg = 'white', lwd = 2.5, cex = 1.5)
-text(2.3, 0.80, 'Process-Guided Deep Learning', pos = 4, cex = 1.1)
+#points_Y <- c(0.79, 0.94, 1.09)
+#pchVal <- 23
+#columns <- c('#7570b3', '#d95f02', '#1b9e77')
+#modelNames <- c('Process-Guided Deep Learning', 'Deep Learning', 'Process-Based')
+#for(i in length(modelNames)){
 
+points(2.2, 0.79, col = '#7570b3', pch = 23, bg = 'white', lwd = 2.5, cex = 1.5)
+text(2.3, 0.8, 'Process-Guided Deep Learning', pos = 4, cex = 1.1)
 points(2.2, 0.94, col = '#d95f02', pch = 22, bg = 'white', lwd = 2.5, cex = 1.5)
 text(2.3, 0.95, 'Deep Learning', pos = 4, cex = 1.1)
-
 points(2.2, 1.09, col = '#1b9e77', pch = 21, bg = 'white', lwd = 2.5, cex = 1.5)
 text(2.3, 1.1, 'Process-Based', pos = 4, cex = 1.1)
 
+#writes it to the png file
 dev.off()
 
-
 # Save the processed data
 readr::write_csv(eval_data, file = file.path(project_output_dir, 'model_summary_results.csv'))
```

Comment:
This should go in a separate function that writes or returns the data object. I think it makes sense to have a separate data munge and plotting function.